hazelcast/src/main/java/com/hazelcast/client/impl/protocol/parameters/ExceptionResultParameters.java

```diff
@@ -33,77 +34,88 @@
      * ClientMessageType of this message
      */
     public static final int TYPE = ResponseMessageConst.EXCEPTION;
+    public int errorCode;
     public String className;
-    public String causeClassName;
     public String message;
-    public String stacktrace;
+    public StackTraceElement[] stackTrace;
+    public int causeErrorCode;
```

Comment:
Having two error codes in an error is confusing for the developers. All a protocol implementer would be interested is the actual cause for the error. Why provide error code if there is a causeErrorCode. May be we can just put the causeErrorCode value in errorCode field and eliminate causeErrorCode field.